code/game/objects/items/devices/scanners.dm

```diff
@@ -537,6 +537,29 @@ GENE SCANNER
 			amount += inaccurate
 	return DisplayTimeText(max(1,amount))
 
+/proc/geigercounter_scan(mob/user, atom/A)
+	var/rad_strength = 0
+	for(var/i in get_rad_contents(A)) // Yes it's intentional that you can't detect radioactive things under rad protection. Gives traitors a way to hide their glowing green rocks.
+		var/atom/thing = i
+		if(!thing)
+			continue
+		var/datum/component/radioactive/radiation = thing.GetComponent(/datum/component/radioactive)
+		if(radiation)
+			rad_strength += radiation.strength
+
+	if(isliving(A))
+		var/mob/living/M = A
+		if(!M.radiation)
+			to_chat(user, "<span class='notice'>[icon2html(A, user)] Radiation levels within normal boundaries.</span>")
+		else
+			to_chat(user, "<span class='boldannounce'>[icon2html(A, user)] Subject is irradiated. Radiation levels: [M.radiation].</span>")
+
+	if(rad_strength)
+		to_chat(user, "<span class='boldannounce'>[icon2html(A, user)] Target contains radioactive contamination. Radioactive strength: [rad_strength]</span>")
+	else
+		to_chat(user, "<span class='notice'>[icon2html(A, user)] Target is free of radioactive contamination.</span>")
+	return TRUE

```

Comment:
Nothing cares about this return value and this proc will never return anything else. 
```suggestion
```